packages/protocol/engine/notarization/manager.go

```diff
@@ -241,11 +243,26 @@ func (m *Manager) createCommitment(index epoch.Index) (success bool) {
 		m.Events.Error.Trigger(errors.Wrap(err, "failed to store latest commitment"))
 	}
 
+	var (
+		createdOutputs = utxo.NewOutputIDs()
+		spentOutputs   = utxo.NewOutputIDs()
+	)
+	m.ledgerState.StateDiffs.StreamCreatedOutputs(index, func(owm *ledger.OutputWithMetadata) error {
```

Comment:
This seems to be very expensive. Can't we put a pointer to `m.ledgerState.StateDiffs.StreamCreatedOutputs` and the consumer (in our case the retainer) can stream it in a separate thread?